presto-main/src/main/java/com/facebook/presto/operator/TableWriterOperator.java

```diff
@@ -158,6 +162,8 @@ public void finish()
     {
         if (state == State.RUNNING) {
             state = State.FINISHING;
+            finishFuture = pageSink.finish();
+            blocked = Futures.allAsList(blocked, toListenableFuture(finishFuture));
```

Comment:
I don't think this is correct. If we have a non-completed future returned from `appendPage()`, this means the sink couldn't accept any more data. However, we told it with `finish()` that there is no more data, and it already started processing the finish request, so we should ignore the old future from `appendPage()`.

Alternatively, we could wait to call `finish()` until the `appendPage()` future is complete, but I don't think that's necessary.